src/nano_contracts/deserializer.ts

```diff
@@ -24,45 +31,58 @@ class Deserializer {
    * We receive these types from the full node, so we
    * use the python syntax
    *
-   * @param {value} Value to deserialize
-   * @param {type} Type of the value to be deserialized
+   * @param {Buffer} buf Value to deserialize
+   * @param {string} type Type of the value to be deserialized
    *
    * @memberof Deserializer
    * @inner
    */
-  deserializeFromType(value: Buffer, type: string): NanoContractArgumentType | null {
-    if (type.endsWith('?')) {
-      // It's an optional
-      const optionalType = type.slice(0, -1);
-      return this.toOptional(value, optionalType);
-    }
-
-    if (type.startsWith('SignedData[')) {
-      return this.toSigned(value, type);
+  deserializeFromType(
+    buf: Buffer,
+    type: string
+  ): DeserializeResult<NanoContractArgumentType | null> {
+    const isContainerType = getContainerType(type) !== null;
```

Comment:
This method calls `getContainerInternalType`, which has the code below

```
  const match = type.match(/^(.*?)\[(.*)\]/);
  const containerType = match ? match[1] : null;
  const internalType = match ? match[2] : null;
  if ((!internalType) || (!containerType)) {
    throw new Error('Unable to extract type');
  }
```

Won't this throw the error when `type` is `str`, for example?

Reply:
I believe we should have tests that ensure this. I couldn't find any tests for `Tuple` or `RawSignedData`, and CodeCov seems to have not found this lack of tests too. :thinking: